src/components/VideoAsset.vue

```diff
@@ -121,8 +134,23 @@ export default {
       : defaultVideoAttributes
     ),
   },
+  watch: {
+    normalisedPosterPath: {
+      immediate: true,
+      handler: 'getPosterDimensions',
+    },
+  },
   methods: {
     normalizeAssetUrl,
+    async getPosterDimensions(path) {
+      let width = null;
+      let height = null;
+      if (path) {
+        ({ width, height } = await getIntrinsicDimensions(path));
+      }
+      this.optimalWidth = width;
+      this.optimalHeight = height;
```

Comment:
I think this `optimalHeight` var is causing some scaling issues in places where the width of the video is larger than the width of the element it is contained in—since you calculate and assign _both_ a with and height attribute to the video element, the height of the video can be way too big in this scenario (since it isn't constrained in the same way that the width is).

To address this, I suggest we simply match what we do for the `<img>` element where we assign the height attribute as `"auto"` when we have a fixed width—that way, we can both assign an ideal width (whether or not it can be used as-is) and the height will scale automatically based on how wide the element actually ends up being (depending on where it ends up in the UI).

I can give you some examples offline of where I'm noticing this issue today.